algoliasearch/client.go

```diff
@@ -258,6 +259,48 @@ func (c *client) BatchWithRequestOptions(operations []BatchOperationIndexed, opt
 	return
 }
 
+func (c *client) WaitTask(indexName string, taskID int) error {
+	return c.WaitTaskWithRequestOptions(indexName, taskID, nil)
+}
+
+func (c *client) WaitTaskWithRequestOptions(indexName string, taskID int, opts *RequestOptions) error {
+	var res TaskStatusRes
+	var err error
+
+	var maxDuration = time.Second
+	var sleepDuration time.Duration
```

Comment:
As you do not reuse the variables outside of the `for` loop, I'd recommend you to remove all those lines, replace the `res, err =  c.GetStatusWithRequestOptions(...)` with  `res, err := c.GetStatusWithRequestOptions(...)` and directly do `sleepDuration := randDuration(time.Second)` in your `for` loop. It would save 5 useless lines.

Reply:
To be honest, I copypasted what was in `index.go`, I didn't even read the function :D